src/Combinatorics/Matroids/realization_space.jl

```diff
@@ -0,0 +1,989 @@
+    R = RS.ambient_ring
+    !(typeof(R) <: MPolyRing) && return RS    
+    Inew = RS.defining_ideal
+    eqs = copy(gens(Inew))
+
+    d = min(dim(Inew), nvars(R))
+    if d == 0
+        nvars(R) == 0 && return RS
+        for p in gens(Inew)
+            f = factor(p)
+            if length(f) > 1
+                push!(eqs,collect(f)[1][1])
+            end
+        end
+    end
+    ineqsnew = RS.inequations
+
+    counter = 0
+    base = 7
+    if RS.char != nothing && RS.char > 0
+        base = RS.char
+    end
+    upperbound = min(base^d, 10^3)
+    while counter < upperbound
+        eqsnew = copy(eqs)
+        values = digits(counter, base=base,pad=d)
+        counter += 1
+
+        for i in 1:d
+            push!(eqsnew, R[i]-values[i])
+        end
+        Inew = ideal(groebner_basis(ideal(R,eqsnew)))
+        isone(Inew) && continue
+        ineqsnew = copy(RS.inequations)
+        for i in 1:length(ineqsnew)
+            ineqsnew[i] = reduce(ineqsnew[i], gens(Inew))
+        end
+        (R(0) in ineqsnew) && continue
+        Inew = stepwise_saturation(Inew,ineqsnew)
+        isone(Inew) && continue
+        break
+    end
+    counter == upperbound && d != 0 && return RS
+
+    RSnew = MatroidRealizationSpace(Inew, ineqsnew, R, RS.realization_matrix, RS.F, RS.char, RS.q)
+    RSnew = reduce_realization_space(RSnew)
+    ineqsnew = RSnew.inequations
+    if length(ineqsnew) > 0
+        Inew = RSnew.defining_ideal
+        Rnew = RSnew.ambient_ring
+        ineqsnew = filter(p ->!isone(ideal(groebner_basis(Inew+ideal(Rnew,p)))),ineqsnew)
+        RSnew = MatroidRealizationSpace(Inew, ineqsnew, Rnew, RSnew.realization_matrix, RSnew.F, RSnew.char, RSnew.q)
+    end
+
+    RSnew.one_realization = true
+
+    return RSnew
+end
+
+
+#####################
+# initial reduction #
+#####################
+
+#=
+function update_hom(v_elim::RingElem, w_repl::RingElem, phi::Oscar.MPolyAnyMap)
+    R = codomain(phi)
+    phi2_im = replace(gens(R), v_elim => w_repl)
+    phi2 = hom(R, R, a->a, phi2_im)
+    return hom(R, R, a->a, phi2.(phi.(gens(R))))
+end
+
+function small_find_solution(v::RingElem, Igens::Vector{<:RingElem})
+
+    with_v_deg_1 = [g for g in Igens if (total_degree(g) == 1 && is_unit(coeff(g,[v],[1])))] 
+    length(with_v_deg_1) == 0 && return "can't isolate"
+    
+    f = first(with_v_deg_1)
+    return -coeff(f, [v], [0])/coeff(f,[v],[1])
+end
+
+
+function small_reduce_one_step(Igens::Vector{<:RingElem}, phi::Oscar.MPolyAnyMap, 
+                               elim::Vector{<:RingElem}, fullyReduced::Bool)
+    Ivars = ideal_vars(Igens); 
+    length(Igens) == 0 && return (Igens, phi, elim, true)
+    for v in Ivars 
+        w = small_find_solution(v, Igens)
+        w isa String && continue
+        
+        push!(elim, v)
+        phi = update_hom(v, w, phi)
+        Igens = filter(x->x!=0, phi.(Igens))
+        return (Igens, phi, elim, false)
+    end
+    return (Igens, phi, elim, true)
+end
+
+
```

Comment:
Remove all this commented code.

Reply:
Removed the initial reduction commented code.